crates/bevy_math/src/cubic_splines.rs

```diff
@@ -1064,22 +1145,42 @@ impl<P: Point> RationalCurve<P> {
         self.iter_samples(subdivisions, Self::acceleration)
     }
 
+    /// Adds a segment to the curve.
     #[inline]
-    /// Adds a segment to the curve
     pub fn push_segment(&mut self, segment: RationalSegment<P>) {
         self.segments.push(segment);
     }
 
     /// Returns the [`RationalSegment`] and local `t` value given a spline's global `t` value.
+    /// Input `t` will be clamped to the domain of the curve. Returned value will be in `[0, 1]`.
     #[inline]
-    fn segment(&self, t: f32) -> (&RationalSegment<P>, f32) {
-        if self.segments.len() == 1 {
-            (&self.segments[0], t)
+    fn segment(&self, mut t: f32) -> (&RationalSegment<P>, f32) {
+        if t <= 0.0 {
+            (&self.segments[0], 0.0)
+        } else if self.segments.len() == 1 {
+            (&self.segments[0], t / self.segments[0].knot_span)
         } else {
-            let i = (t.floor() as usize).clamp(0, self.segments.len() - 1);
-            (&self.segments[i], t - i as f32)
+            // Try to fit t into each segment domain
+            for segment in self.segments.iter() {
+                if t < segment.knot_span {
+                    // The division here makes t a normalized parameter in [0, 1] that can be properly
+                    // evaluated against a cubic curve segment. See equations 6 & 16 from "Matrix representation
+                    // of NURBS curves and surfaces" by Choi et al. or equation 3 from "General Matrix
+                    // Representations for B-Splines" by Qin.
+                    return (segment, t / segment.knot_span);
+                } else {
+                    t = t - segment.knot_span
```

Comment:
This line causes CI to error.
```suggestion
                    t -= segment.knot_span;
```